Brokerages/GDAX/GDAXBrokerage.cs

```diff
@@ -0,0 +1,349 @@
+        }
+
+        /// <summary>
+        /// Creates a new order
+        /// </summary>
+        /// <param name="order"></param>
+        /// <returns></returns>
+        public override bool PlaceOrder(Orders.Order order)
+        {
+            var req = new RestRequest("/orders", Method.POST);
+
+            dynamic payload = new ExpandoObject();
+
+            payload.size = Math.Abs(order.Quantity);
+            payload.side = order.Direction.ToString().ToLower();
+            payload.type = ConvertOrderType(order.Type);
+            payload.price = order is LimitOrder ? ((LimitOrder)order).LimitPrice : order is StopMarketOrder ? ((StopMarketOrder)order).StopPrice : 0;
+            payload.product_id = ConvertSymbol(order.Symbol);
+
+            if (_algorithm.BrokerageModel.AccountType == AccountType.Margin)
+            {
+                payload.overdraft_enabled = true;
+            }
+
+            req.AddJsonBody(payload);
+
+            GetAuthenticationToken(req);
+            var response = RestClient.Execute(req);
+
+            if (response.StatusCode == System.Net.HttpStatusCode.OK && response.Content != null)
+            {
+                dynamic raw = JsonConvert.DeserializeObject<dynamic>(response.Content);
+
+                if (raw == null || raw.id == null)
+                {
+                    Log.Error("GDAXBrokerage.PlaceOrder: Error parsing response from place order");
+
+                    return false;
+                }
+
+                string brokerId = raw.id;
+                if (CachedOrderIDs.ContainsKey(order.Id))
+                {
+                    CachedOrderIDs[order.Id].BrokerId.Add(brokerId);
+                }
+                else
+                {
+                    order.BrokerId.Add(brokerId);
+                    CachedOrderIDs.TryAdd(order.Id, order);
+                }
+
+                if (order.Type != OrderType.Market)
+                {
+                    FillSplit.TryAdd(order.Id, new GDAXFill(order));
+                }
+
+                OnOrderEvent(new OrderEvent(order, DateTime.UtcNow, 0, "GDAX Order Event") { Status = OrderStatus.Submitted });
+
+                if (order.Type == OrderType.Market)
+                {
+                    OnOrderEvent(new OrderEvent(order, DateTime.UtcNow, (decimal)raw.fill_fees, "GDAX Order Event") { Status = OrderStatus.Filled });
+                    Orders.Order outOrder = null;
+                    CachedOrderIDs.TryRemove(order.Id, out outOrder);
+                }
+
+                Log.Trace("GDAXBrokerage.PlaceOrder: Order completed successfully orderid:" + order.Id.ToString());
+                return true;
+
+            }
+
+            OnOrderEvent(new OrderEvent(order, DateTime.UtcNow, 0, "GDAX Order Event") { Status = OrderStatus.Invalid });
+            Log.Trace("GDAXBrokerage.PlaceOrder: Order failed Order Id: " + order.Id + " timestamp:" + order.Time + " quantity: " + order.Quantity.ToString()
```

Comment:
Log.Error for failed error messages; & probably returning some information to the user-side via brokerage messaging handler.